internal/k8s/endpoint_slices.go

```diff
@@ -0,0 +1,20 @@
+package k8s
+
+import (
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+)
+
+type EpsOrSlices struct {
+	Type      EpsOrSliceType
+	EpVal     *v1.Endpoints
+	SlicesVal []*discovery.EndpointSlice
+}
+
+type EpsOrSliceType int
+
+const (
+	Unknown EpsOrSliceType = iota
```

Comment:
Do we need an "Unknown" case?

Reply:
I think (but it's been some time) I added this when I had some failures in tests because I forgot to change the type. I thought being explicit about the values instead of having a meaningful 0-value was better.

Reply:
Ok, so you think it is better to be in?

Reply:
Well, now that we have unit tests maybe it's not necessary. It would be better to have it explode than silently skipping an if because we forgot to set the right type somewhere.

I will remove it.

Reply:
I think I remember (or, the unit test reminded me).
With this extra value, when we create an empty `EpsOrSlices{}` we know both fields are nil.
On the other hand, with the default value being Eps (or slices), we would need to initialize the field (or have not nil checks around the code).

I'd leave it there.